contracts/apm/Repo.sol

```diff
@@ -0,0 +1,97 @@
+pragma solidity ^0.4.15;
+
+import "../apps/AragonApp.sol";
+
+
+contract Repo is AragonApp {
+    struct Version {
+        uint16[3] semanticVersion;
+        address contractAddress;
+        bytes contentURI;
+    }
+
+    Version[] versions;
+    mapping (bytes32 => uint256) versionIdForSemantic;
+    mapping (address => uint256) latestVersionIdForContract;
+
+    bytes32 constant public CREATE_VERSION_ROLE = bytes32(1);
+
+    event NewVersion(uint256 versionId, uint16[3] semanticVersion);
+
+    /**
+    * @notice Create new version for repo
+    * @param _newSemanticVersion Semantic version for new repo version
+    * @param _contractAddress address for smart contract logic for version (if set to 0, it uses last versions' contractAddress)
+    * @param _contentURI External URI for fetching new version's content
+    */
+    function newVersion(
+        uint16[3] _newSemanticVersion,
+        address _contractAddress,
+        bytes _contentURI
+    ) auth(CREATE_VERSION_ROLE) public
+    {
+        address contractAddress = _contractAddress;
+        if (versions.length > 0) {
+            Version storage lastVersion = versions[versions.length - 1];
+            require(isValidBump(lastVersion.semanticVersion, _newSemanticVersion));
+            if (contractAddress == 0)
+                contractAddress = lastVersion.contractAddress;
+            // Only allows smart contract change on major version bumps
+            require(lastVersion.contractAddress == contractAddress || _newSemanticVersion[0] > lastVersion.semanticVersion[0]);
+        } else {
+            versions.length += 1;
```

Comment:
Dont understand this

Reply:
As I'm understanding it, if there was no previous version listed, we using `version[1]` to hold the first version.

`version[0]` is then always `0.0.0` (which would be the "first" version, since it's the first in the array).